server/src/main/java/org/elasticsearch/transport/TransportActionProxy.java

```diff
@@ -73,7 +93,7 @@ public void handleException(TransportException exp) {
                 @Override
                 public TransportResponse read(StreamInput in) throws IOException {
                     if (in.getTransportVersion().equals(channel.getVersion()) && in.supportReadAllToReleasableBytesReference()) {
-                        return new BytesTransportResponse(in.readAllToReleasableBytesReference());
+                        return new BytesTransportResponse(in.readAllToReleasableBytesReference(), channel.getVersion());
```

Comment:
nit (these things are checked equal today, but in principle we could weaken that check in future):

```suggestion
                        return new BytesTransportResponse(in.readAllToReleasableBytesReference(), in.getTransportVersion());
```